packages/sanity/src/desk/structureBuilder/Intent.ts

```diff
@@ -9,10 +9,10 @@ import {StructureNode} from './StructureNodes'
 export type IntentJsonParams = {[key: string]: any}
 
 /**
- * Type for base intent parameters
+ * Interface for base intent parameters
  *
  * @public */
-export type BaseIntentParams = {
```

Comment:
Looked at the reason why the typecheck is failing and it seems to be cause of this change from `type` to `interface` My guess is because 

`export type IntentParams = BaseIntentParams | [BaseIntentParams, IntentJsonParams]` 

and `export type IntentParameters =
  | Record<string, unknown>
  | [Record<string, unknown>, Record<string, unknown>]` are same when the sub type is a type but if it's an interface it's not equal. I don't know why it does it but someone with more TS experience might know